oas_docs/README.md

```diff
@@ -1,9 +1,164 @@
 
-To apply some overlays that perform some post-processing and append some content, run `make api-docs-overlay`.
\ No newline at end of file
+## Workflow
+
+The workflow consists of two steps where the final goal is to produce Kibana OpenAPI bundle containing Kibana's API endpoints we expect to be published on the API reference documentation platform.
+
+ESS and Serverless OpenAPI bundles generation workflows are pretty similar so an there is a common workflow is depicted below
+
+![alt text](workflow.png)
+
+### Step 1
+
+At the first step domain OpenAPI bundles are produces. It depends on solution/team what approach to choose. There are three ways to get a domain OpenAPI bundle
+
+- Use **Specification first approach** via [`kbn-openapi-bundler`](../packages/kbn-openapi-bundler/README.md) to process source OpenAPI specs describing individual OpenAPI endpoints. It works well with OpenAPI code generation and provides a lot of flexibility during development to avoid publishing documentation for not yet released API endpoints.
+
+- Use **Code first approach** via `scripts/capture_oas_snapshot.js` to capture OpenAPI specs from routes source code. It allows to produce a domain bundle with minimum effort but requires running ES and Kibana for that and makes it hard to control what documentation is published eventually.
+
+- Use **Manually written** domain OpenAPI bundles if neither of the above doesn't work. It's considered as a temporary measure and highly discouraged as a long term approach.
+
+### Step 2
+
+At the second step domain OpenAPI bundles are merged together by `kbn-openapi-bundler` to produce the result bundle.
+
+### Scripts
+
+`scripts` folder contains scripts describing source domain OpenAPI bundles, result document configuration and output path. There are two groups currently
+
+- `merge_ess_oas.js` and `merge_serverless_oas.js` to produce production ready Kibana OpenAPI bundles for ESS and Serverless respectively.
+
+- `merge_ess_oas_staging.js` and `merge_serverless_oas_staging.js` to produce staging Kibana OpenAPI bundles for ESS and Serverless respectively. Staging bundles currently aren't published anywhere and required for polishing API reference documentation on domains not yet included in the production ready Kibana OpenAPI bundles.
+
+### Output Kibana OpenAPI bundles
+
+`output` folder contains result Kibana OpenAPI bundles. There are two groups currently
+
+- `kibana.yaml` and `kibana.serverless.yaml` are production ready ESS and Serverless Kibana OpenAPI bundles to be published on the API reference documentation platform. Publishing process isn't fully automated yet and performed by the Docs Engineering team.
+
+- `kibana.staging.yaml` and `kibana.serverless.staging.yaml` are staging ESS and Serverless Kibana OpenAPI bundles. Staging bundles currently aren't published anywhere and required for polishing API reference documentation on domains not yet included in the production ready Kibana OpenAPI bundles.
```

Comment:
I think we shouldn't repeat ourselves, the purpose of these bundles have been already described above in the Scripts section. Perhaps we could combine these sections into one to avoid duplication.